packages/flutter_tools/test/src/testbed.dart

```diff
@@ -61,11 +80,19 @@ class Testbed {
   final Map<Type, Generator> _overrides;
 
   /// Runs `test` within a tool zone.
-  FutureOr<T> run<T>(FutureOr<T> Function() test) {
+  ///
+  /// `overrides` may be used to provide new context values for the single test
+  /// case or override any context values from the setup.
+  FutureOr<T> run<T>(FutureOr<T> Function() test, {Map<Type, Generator> overrides}) {
     final Map<Type, Generator> testOverrides = Map<Type, Generator>.from(_testbedDefaults);
+    // Add the initial setUp overrides
     if (_overrides != null) {
       testOverrides.addAll(_overrides);
     }
+    // Add the test-specific overrides
```

Comment:
Some tests that use 'is' checks to make sure that the overrides work as documented would be good. What will/should happen with different combos of overrides appearing {in the default list above, in the Testbed constructor, in the run call}?

Reply:
Added tests